### PR TITLE
Should fix mining complexes landing on the odd mining areas, such as metaclubs mining dock

### PR DESCRIPTION
**Pull request by MadmanMartian**
Should stop angies lair eating mining on some maps.

### CODE REVIEW

File: code/game/asteroid.dm [hunk 1]
```diff
@@ -37,8 +37,8 @@ proc/check_complex_placement(var/turf/T,var/size_x,var/size_y,var/ignore_walls=0
 	surroundings |= range(7, locate(T.x,T.y+size_y,T.z))
 	surroundings |= range(7, locate(T.x+size_x,T.y+size_y,T.z))
 
-	for(var/area in list(/area/mine/explored,/area/vox_trading_post,/area/research_outpost, /area/asteroid, /area/asteroid/clown))
-		if(locate(area) in surroundings)// +5s are for view range
+	for(var/area/A in surroundings)
+		if(!istype(A,/area/mine/unexplored))
 			return 0
 
 	if(locate(/turf/space) in surroundings)
```
Comment (Kurfursten) [inline code/game/asteroid.dm:44]: This next check should(??) no longer be necessary because it is already skipping the space area.

Comment (MadmanMartian) [reply code/game/asteroid.dm:44]: Space turfs can still be within actual areas.

Comment (Kurfursten) [reply code/game/asteroid.dm:44]: They can be, but I'm trying to imagine where they happens on any of our maps. Maybe the high yield asteroids?

Comment (MadmanMartian) [reply code/game/asteroid.dm:44]: Call it futureproofing, in case of a bad mapper, or a weird map

**Pull request merged**